crypto/types/cache.go

```diff
@@ -0,0 +1,83 @@
+package types
+
+import (
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+const (
+	TxHashLen        = 32
+	AddressStringLen = 2 + 20*2
+)
+
+// Cache is a cache of verified signatures
+type Cache struct {
+	data *lru.Cache[string, []byte]
+}
+
+// NewSignatureCache initializes the signature cache.
+// signature verification is one of the most expensive parts in verification
+// by caching it we avoid needing to verify the same signature multiple times
+func NewSignatureCache() *Cache {
+	// 500 * (32 + 42) = 37.5KB
+	cache, err := lru.New[string, []byte](500)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
nit: return the error and then let the caller panic if desired. Generally, I've been trying to push for avoiding panics in constructors, but if we must panic, panic in the caller.